daemon/entrypoint.sh

```diff
@@ -276,7 +276,12 @@ function osd_directory {
   fi
 
   for OSD_ID in $(ls /var/lib/ceph/osd |  awk 'BEGIN { FS = "-" } ; { print $2 }'); do
-    if [ -n "${JOURNAL_DIR}" ]; then
+    OSD_JOURNAL_TMP="OSD_JOURNAL$OSD_ID"
+    OSD_JOURNAL=${!OSD_JOURNAL_TMP}
+    if [ -n "${OSD_JOURNAL}" ]; then
+      ceph-disk -v prepare ${OSD_JOURNAL}
```

Comment:
Incidentally, I do have another branch I am using on my systems that simply adds the ability to pass a flag to "preserve journals", which really just skips the `ceph-disk prepare` step. I haven't submitted it because I've only added it for osd_directory right now, as that is the only osd mode I use in ceph-docker.

diff --git c/daemon/README.md w/daemon/README.md
index f31e94c..73d6542 100644
--- c/daemon/README.md
+++ w/daemon/README.md
@@ -237,6 +237,7 @@ following environment variables will control its creation:
 * `JOURNAL` is the location of the journal (default is the `journal` file inside the OSD data directory)
 * `HOSTNAME` is the name of the host; it is used as a flag when adding the OSD to the CRUSH map
 * `OSD_JOURNAL<osd-id>` optional - providing for each OSD directory to be used will prepare and mount the supplied block device
+* `JOURNAL_PRESERVE` optional - if set, skips the prepare step for OSD_JOURNAL

 The old option `OSD_ID` is now unused.  Instead, the script will scan for each directory in `/var/lib/ceph/osd` of the form `<cluster>-<osd-id>`.

diff --git c/daemon/entrypoint.sh w/daemon/entrypoint.sh
index 8a0b5eb..5a4bacd 100755
--- c/daemon/entrypoint.sh
+++ w/daemon/entrypoint.sh
@@ -279,7 +279,9 @@ function osd_directory {
     OSD_JOURNAL_TMP="OSD_JOURNAL$OSD_ID"
     OSD_JOURNAL=${!OSD_JOURNAL_TMP}
     if [ -n "${OSD_JOURNAL}" ]; then
-      ceph-disk -v prepare ${OSD_JOURNAL}
+      if [ -z "${JOURNAL_PRESERVE}" ]; then
+        ceph-disk -v prepare ${OSD_JOURNAL}
+      fi
       OSD_J=${OSD_JOURNAL}
     elif [ -n "${JOURNAL_DIR}" ]; then
        OSD_J="${JOURNAL_DIR}/journal.${OSD_ID}"